<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>PenRed Geometry Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="autoFillBackground">
    <bool>true</bool>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="20,1">
    <property name="leftMargin">
     <number>5</number>
    </property>
    <property name="topMargin">
     <number>5</number>
    </property>
    <property name="rightMargin">
     <number>5</number>
    </property>
    <property name="bottomMargin">
     <number>5</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_9" stretch="1">
      <property name="spacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_images" stretch="0,0">
        <property name="spacing">
         <number>4</number>
        </property>
        <item>
         <spacer name="horizontalSpacer_left">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <spacer name="horizontalSpacer_right">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_8" stretch="0,0">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="keyTab">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <attribute name="title">
          <string>Key</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QTextEdit" name="keyText">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <attribute name="title">
          <string>Tests</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QTextEdit" name="testOutput">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="testButton">
            <property name="text">
             <string>Test</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="QScrollArea" name="viewerParameters">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>2</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollArea">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>262</width>
           <height>767</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <property name="sizeConstraint">
             <enum>QLayout::SetDefaultConstraint</enum>
            </property>
            <item>
             <widget class="QLabel" name="label_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>View orientation:</string>
              </property>
              <property name="margin">
               <number>3</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="perspectiveSelector">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>View type:</string>
              </property>
              <property name="margin">
               <number>3</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="matBodyViewButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Mat</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,0,0,0">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Position (x,y,z) cm:</string>
              </property>
              <property name="margin">
               <number>6</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="Xedit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>0.0e0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="Yedit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>0.0e0</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="Zedit">
              <property name="sizePolicy">
               <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>0.0e0</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QTabWidget" name="parametersInput">
            <property name="currentIndex">
             <number>0</number>
            </property>
            <widget class="QWidget" name="tab2D">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <attribute name="title">
              <string>2D</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_6">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_12" stretch="0,0,0,0,0,0">
                <property name="sizeConstraint">
                 <enum>QLayout::SetDefaultConstraint</enum>
                </property>
                <item>
                 <widget class="QLabel" name="label_5">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string>Resolution (pixels) :</string>
                  </property>
                  <property name="margin">
                   <number>5</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="resolutionEditX">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="minimum">
                   <number>10</number>
                  </property>
                  <property name="maximum">
                   <number>2000</number>
                  </property>
                  <property name="value">
                   <number>600</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="resolutionEditY">
                  <property name="minimum">
                   <number>10</number>
                  </property>
                  <property name="maximum">
                   <number>2000</number>
                  </property>
                  <property name="value">
                   <number>600</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLabel" name="label_6">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="text">
                   <string> Pixel Size (cm): </string>
                  </property>
                  <property name="margin">
                   <number>5</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="pixelSizeEdit">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="wrapping">
                   <bool>false</bool>
                  </property>
                  <property name="decimals">
                   <number>10</number>
                  </property>
                  <property name="minimum">
                   <double>0.000000000000000</double>
                  </property>
                  <property name="maximum">
                   <double>100.000000000000000</double>
                  </property>
                  <property name="stepType">
                   <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                  </property>
                  <property name="value">
                   <double>0.100000000000000</double>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
            <widget class="QWidget" name="tab3D">
             <attribute name="title">
              <string>3D</string>
             </attribute>
             <layout class="QVBoxLayout" name="verticalLayout_7">
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_10">
                <item>
                 <widget class="QLabel" name="label_19">
                  <property name="text">
                   <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Camera ( ρ , θ , φ ) : &lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                  </property>
                  <property name="margin">
                   <number>6</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="rhoEdit">
                  <property name="text">
                   <string>10.0</string>
                  </property>
                  <property name="placeholderText">
                   <string>10.0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="thetaEdit">
                  <property name="text">
                   <string>1.571</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="phiEdit">
                  <property name="text">
                   <string>0.0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_11">
                <item>
                 <widget class="QLabel" name="label_10">
                  <property name="text">
                   <string>Camera (x, y, z) cm:</string>
                  </property>
                  <property name="margin">
                   <number>6</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lookX">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="text">
                   <string>0.0</string>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="placeholderText">
                   <string>0.0e0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lookY">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="text">
                   <string>0.0</string>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="placeholderText">
                   <string>0.0e0</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="lookZ">
                  <property name="text">
                   <string>0.0</string>
                  </property>
                  <property name="readOnly">
                   <bool>true</bool>
                  </property>
                  <property name="placeholderText">
                   <string>1.0e0</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_14">
                <item>
                 <widget class="QLabel" name="label_14">
                  <property name="text">
                   <string>Resolution (pixels):</string>
                  </property>
                  <property name="margin">
                   <number>6</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="resolutionH3D">
                  <property name="minimum">
                   <number>8</number>
                  </property>
                  <property name="maximum">
                   <number>1000</number>
                  </property>
                  <property name="value">
                   <number>400</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QSpinBox" name="resolutionV3D">
                  <property name="minimum">
                   <number>8</number>
                  </property>
                  <property name="maximum">
                   <number>1000</number>
                  </property>
                  <property name="value">
                   <number>400</number>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout_15">
                <item>
                 <widget class="QLabel" name="label_17">
                  <property name="text">
                   <string>Pixel Size (cm):</string>
                  </property>
                  <property name="margin">
                   <number>6</number>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QDoubleSpinBox" name="pixelSize3D">
                  <property name="decimals">
                   <number>10</number>
                  </property>
                  <property name="maximum">
                   <double>100.000000000000000</double>
                  </property>
                  <property name="stepType">
                   <enum>QAbstractSpinBox::AdaptiveDecimalStepType</enum>
                  </property>
                  <property name="value">
                   <double>0.100000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuLoad">
     <property name="title">
      <string>Load</string>
     </property>
     <addaction name="actionconfig"/>
     <addaction name="actionQadric"/>
     <addaction name="actionMesh"/>
    </widget>
    <addaction name="separator"/>
    <addaction name="menuLoad"/>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuViews">
    <property name="title">
     <string>Views</string>
    </property>
    <addaction name="actionAdd"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionColors"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuViews"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionconfig">
   <property name="text">
    <string>Configuration</string>
   </property>
  </action>
  <action name="actionQadric">
   <property name="text">
    <string>Quadric</string>
   </property>
  </action>
  <action name="actionMesh">
   <property name="text">
    <string>Mesh</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionAdd">
   <property name="text">
    <string>Add</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>Delete</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionColors">
   <property name="text">
    <string>Colors</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
